native-stack/package.json

```diff
@@ -1,6 +1,5 @@
 {
-  "main": "../lib/commonjs/native-stack/index",
   "module": "../lib/module/native-stack/index",
   "react-native": "../src/native-stack/index",
-  "types": "../lib/typescript/native-stack/index"
+  "types": "../lib/typescript/native-stack/index.d.ts"
```

Comment:
I've added file formats because react-native-builder bob (while using the newest version) throws an error while the file format is not specified.

Reply:
I've removed it, as I was getting weird errors about unknown `_$_REQUIRE` functions while trying to launch an app.
This was happening on the first times when I was moving the app to ESM and that error does not happen again, so I believe we could bring the `main` back 👍

Reply:
About the `.d.ts` addition - somehow react-native-builder-bob requires right now to have that specific file format, as it can't find file by `index` shortcut name anymore.

Reply:
Reverting `main` field - resolved in https://github.com/software-mansion/react-native-screens/pull/1982/commits/30873a08b3c2686c2dbbc848b29aa8994d81a909.

Reply:
Small update about adding `.d.ts` format here - I see that some package managers were not detecting `index` file properly, so probably in react-native-builder-bob there's a requirement to do such file format 👍 

https://publint.dev/react-native-screens@3.27.0